File: September'22/258-add-digits/258-add-digits.cpp
class Solution {
public:
    int add(int n){
            int sum = 0;
            while (n > 0){
                sum+=(n%10);
                n /= 10;
            }
            return sum;
        }
    int addDigits(int num) {
        
        if(num == 0){
            return num;
        }
        while(num>9){
            num = add(num);
        }
        return num;
    }
};

File: September'22/258-add-digits/258-add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        def add(n):
            sum = 0
            while n > 0:
                sum += (n%10)
                n//=10
            return sum
        if num == 0:
            return num
        while num > 9:
            num = add(num)
        return num